packages/abi-coder/src/json-abi.ts

```diff
@@ -58,10 +73,16 @@ export type JsonAbi = ReadonlyArray<JsonAbiFragment> | JsonFlatAbi;
 export class ABI {
   readonly types: ReadonlyArray<JsonFlatAbiFragmentType>;
   readonly functions: ReadonlyArray<JsonFlatAbiFragmentFunction>;
+  readonly loggedTypes: ReadonlyArray<JsonFlatAbiFragmentLoggedType>;
 
   constructor(jsonAbi: JsonFlatAbi) {
     this.types = jsonAbi.types;
     this.functions = jsonAbi.functions;
+    this.loggedTypes = jsonAbi.loggedTypes;
+  }
+
+  parseLoggedType(loggedType: JsonFlatAbiFragmentLoggedType): JsonAbiFragmentType {
+    return ParamType.fromObject(this.parseInput(loggedType.loggedType) as JsonFragmentType);
```

Comment:
It was much easier to reuse the current logic - but this should be cleaned up asap